File: Module 07/Lab/StarWars_Common/src/main/java/edu/neumont/interfaces/IRepository.java
package edu.neumont.interfaces;


import edu.neumont.dtos.DtoBase;

import java.util.List;

public interface IRepository<T extends DtoBase>
{
    List<T> get();
    T get(String id);
    void save(T target);
    void update(String id, T target);
    void delete(String id);
}

File: Module 07/dotNet/StarWars.Common/StarWars.Common/Interfaces/IRepository.cs
using System;
using System.Collections.Generic;
using StarWars.Common.Dtos;

namespace StarWars.Common.interfaces
{
    public interface IRepository<T> where T: DtoBase
    {
        List<T> Get();
        T Get(string id);
        void Save(T target);
        void Update(string id, T target);
        void Delete(string id);
    }
}